src/FffPolygonGenerator.cpp

```diff
@@ -680,6 +681,12 @@ void FffPolygonGenerator::processSkinsAndInfill(SliceMeshStorage& mesh, unsigned
         }
         generateInfill(layer_nr, mesh, innermost_wall_line_width, infill_skin_overlap, wall_line_count);
     }
+
+    if (mesh.getSettingBoolean("ironing_enabled"))
+    {
+        TopSurface* top_surface = new TopSurface(mesh, layer_nr); //Generate the top surface to iron over.
```

Comment:
You initialize this on the heap, but don't delete it.

Please revert https://github.com/Ultimaker/CuraEngine/pull/534/commits/6d1020f37536d037a45fcea7824a4ee37fa04daf